### PR TITLE
[docs] Added Rejection Reasons document

### LINKED ISSUE #2742: Dedicated document for Rejection Reason Codes
Currently rejection reason codes are described in several places.

- [API-functions.md#rejection-reasons](https://github.com/Haivision/srt/blob/master/docs/API/API-functions.md#rejection-reasons)
- [Access Control](https://github.com/Haivision/srt/blob/master/docs/features/access-control.md#rejection-codes)

Better to have a separate document, especially given that more might be defined further and a more extensive use may be expected (e.g. see #1292, #2638).

## TODO

Create a dedicated document in `docs/API/rejection-codes.md` and compile the codes from both places mentioned above.
Add references to the new document.

- [ ] `SRT_REJ_CRYPTO` is missing from [API-functions.md#rejection-reasons-1](https://github.com/Haivision/srt/blob/master/docs/API/API-functions.md#rejection-reasons-1)

### PR DESCRIPTION
**Pull request by maxsharabayko**
Fixes #2742.

### CODE REVIEW

File: docs/API/rejection-codes.md [hunk 1]
```diff
@@ -0,0 +1,470 @@
+# SRT Rejection Codes
+
+This document provides an overview of the rejection (error) codes used by and supported within SRT and SRT-based applications. 
+
+[:arrow_down: &nbsp; Jump to list of rejection codes](#api-function-rejection-codes)
+
+
+## Summary of Rejection Codes
+
+Rejection codes are used in the SRT API, and are transferred on the wire as a part of a Handshake packet (refer to the `Handshake Type` field of the Handshake packet).
```
Comment (stevomatthews) [inline docs/API/rejection-codes.md:10]:
```suggestion
Rejection codes are used in the SRT API, and are transferred on the wire as a part of a Handshake packet (refer to the `Handshake Type` field of the [Handshake](./docs/features/handshake.md) packet).
```

File: docs/API/rejection-codes.md [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+| 1505   | [SRT_REJX_VERSION](#SRT_REJX_VERSION)             | 1.4.2   | SRT application version not supported.                                                                             |
+| 1507   | [SRT_REJX_NOROOM](#SRT_REJX_NOROOM)               | 1.4.2   | Data stream cannot be archived due to lack of storage space.                                                   |
+|        | <img width=250px height=1px/>                     |         |                                                                                                                |
+
+
+**NOTE**: SRT rejection codes follow this prefix convention:
+
+  - `SRT_REJ`: standard rejection codes from SRT API functions (0 - 99)
+  - `SRT_REJC`: mark the border values between ranges.
+  - `SRT_REJX`: extended rejection codes (400 – 599)  :warning: *@max is this the correct range (see comments above)?*
```
Comment (stevomatthews) [inline docs/API/rejection-codes.md:127]: See inline comment.

File: docs/API/rejection-codes.md [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+the `access_control.h` public header file. The intention is that applications
+using these codes understand the situations they describe in a standard way.
+
+- `SRT_REJC_USERDEFINED` and above: These are errors that are freely defined by 
+the application. Codes from this range can be only understood if each application 
+knows the code definitions of the other. These codes should be used only after 
+making sure that the applications at either end of a connection understand them.
+
+The intention here is for the predefined codes to be consistent with the HTTP
+standard codes. Such code can be set by using the `srt_setrejectreason` function.
```
Comment (stevomatthews) [inline docs/API/rejection-codes.md:274]:
```suggestion
standard codes. Such code can be set by using the [`srt_setrejectreason`](docs/api/api-functions.md#srt-setrejectreason) function.
```

File: docs/API/rejection-codes.md [hunk 4]
```diff
@@ -0,0 +1,470 @@
+# SRT Rejection Codes
+
+This document provides an overview of the rejection (error) codes used by and supported within SRT and SRT-based applications.
```
Comment (stevomatthews) [inline docs/API/rejection-codes.md:3]:
```suggestion
This document provides an overview of the rejection (error) codes used by and supported within SRT and SRT-based applications. For information on other types of error codes refer to the [API Socket Options](./docs/API/API-socket-options.md) document.
```

**Review by stevomatthews — COMMENTED** [review]
Added some links, plus minor edits

File: docs/API/rejection-codes.md [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+  - `SRT_REJC_INTERNAL` = 0
+  - `SRT_REJC_PREDEFINED` = 1000
+  - `SRT_REJC_USERDEFINED` = 2000
+  
+When transferred on the wire, the API value is incremented by 1000 to become the `Handshake Type` field value. In the following sections the values of rejection reason codes are given in accordance with the API values.
+
+
+### SRT Internal Rejection Codes
+
+Defined in [**srt.h**](srtcore/srt.h), these codes provide the reason why a connection is rejected by SRT. They cover the reserved range 0 - 999 (below `SRT_REJC_PREDEFINED`).
```
Comment (ethouris) [inline docs/API/rejection-codes.md:28]:
```suggestion
Defined in [**srt.h**](srtcore/srt.h), these codes provide the reason why a connection is rejected by SRT. They cover the reserved range 0 - 999 (below `SRT_REJC_PREDEFINED`). These codes cannot be used by applications to report a rejection reason.
```

File: docs/API/rejection-codes.md [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+
+Naming: `SRT_REJ_*`
+
+  - `SRT_REJ_UNKNOWN` = 0
+  - `SRT_REJ_SYSTEM` = 1
+  - ...
+  - `SRT_REJ_CRYPTO` = 17
+
```
Comment (ethouris) [inline docs/API/rejection-codes.md:36]:
```suggestion
See [the list below](#api-function-rejection-codes) for details.
```

File: docs/API/rejection-codes.md (hunk 2 above)
Comment (ethouris) [reply docs/API/rejection-codes.md:127]: Should be simply "code values above 1000", if you speak about the `SRT_REJX` prefix.

File: docs/API/rejection-codes.md [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+
+**NOTE**: SRT rejection codes follow this prefix convention:
+
+  - `SRT_REJ`: standard rejection codes from SRT API functions (0 - 99)
+  - `SRT_REJC`: mark the border values between ranges.
+  - `SRT_REJX`: extended rejection codes (400 – 599)  :warning: *@max is this the correct range (see comments above)?*
+
+
+## API Function Rejection Reasons
```
Comment (ethouris) [inline docs/API/rejection-codes.md:130]: Wasn't exactly the same already collected in a table above?

File: docs/API/rejection-codes.md [hunk 8]
```diff
@@ -0,0 +1,470 @@
...
+
+Internal Program Error. This should not happen during normal usage. It
+usually indicates a bug in the software (although this can be reported by both
+local and foreign hosts).
+
+  
+#### SRT_REJ_CLOSE
+
+The listener socket was able to receive the request, but is currently
+ being closed. It's likely that the next request will result in a timeout.
```
Comment (ethouris) [inline docs/API/rejection-codes.md:179]: leading space

Comment (maxsharabayko) [reply docs/API/rejection-codes.md:179]:
```suggestion
being closed. It's likely that the next request will result in a timeout.
```

**Pull request merged**